docs/MOTD.md

```diff
@@ -0,0 +1,362 @@
+# MOTD Protocol Documentation
+
+## Overview
+
+The server operates by listening on port `19132` for incoming RakNet unconnected pings. Upon receiving a valid ping, the server responds with an unconnected pong that contains the MOTD and other info. This interaction ensures that clients receive essential details about the server before establishing a connection.
+
+## Packet Structure
+
+### Unconnected Ping
+
+When a client wants to display the server it sends an unconnected ping
+
+- **Packet ID (1 byte):** `0x01`
+- **Timestamp (8 bytes, Big-Endian):** The time since start of the client. Just setting it to current unix timestamp works fine
+- **Magic Offline Message ID (16 bytes):**
+  ```
+  [
+      0x00, 0xFF, 0xFF, 0x00, 
+      0xFE, 0xFE, 0xFE, 0xFE, 
+      0xFD, 0xFD, 0xFD, 0xFD, 
+      0x12, 0x34, 0x56, 0x78
+  ]
+  ```
+- **GUID (8 bytes, Big-Endian):** Unique identifier for the client.
+
+#### Validation Criteria
+
+1. **Packet Size:**  
+   The total size must be exactly 33 bytes.
+2. **Packet ID:**  
+   Must strictly be `0x01`.
+3. **Timestamp:**  
+   It is just a timestamp of uptime, but we can check it is not infront of unix timestamp
+4. **Magic Offline Message ID:**  
+   Must match the predefined 16-byte sequence.
+5. **GUID:**  
+   It doesn't really need to be checked, but it is there
+
+### Unconnected Pong
+
+In response to a valid unconnected ping, the server dispatches an unconnected pong packet containing the MOTD and additional server details.
+
+- **Packet ID (1 byte):** `0x1C` (Denotes Unconnected Pong)
+- **Timestamp (8 bytes, Big-Endian):** The time since start of the server. Just setting it to current unix timestamp works fine
+- **Server GUID (8 bytes, Big-Endian):** Unique server identifier.
+- **Magic Offline Message ID (16 bytes):** Identical to the one used in the ping.
+- **MOTD Length (2 bytes, Big-Endian):** Specifies the length of the MOTD string.
+- **MOTD (Variable Length):** The actual Message of the Day in byte format.
+
+## Message of the Day (MOTD) Generation
+
+The MOTD is what shows up in the server list, it is what people see before joining the server.
+
+### MOTD Format
+
+The MOTD follows a semicolon-separated (`;`) format comprising various fields, semi colons can be escaped with a backslash (`\`)
+```
+<Edition>;<MOTD>;<Protocol Version>;<Game Version>;<Player Count>;<Max Player Count>;<Server GUID>;<World Name>;<Game Mode>;<Nintendo Limited>;<IpV4Port>;<IpV6Port>;<Extra Data>;
+```
+
+#### Field Breakdown & Data Types
+
+- **Edition:** string  
+  Specifies the game edition (e.g., `MCPE` for Minecraft Pocket Edition), seems to be hardcoded to `MCPE`
+- **MOTD:** string  
+  The actual message that is displayed, it can have Minecraft colors.  
+- **Protocol Version:** uint16  
+  Numeric identifier indicating the game protocol version (e.g., `748`).  
+- **Game Version:** string  
+  Denotes the server's game version (e.g., `1.20.40`, or on CubeCraft: `1`).  
```

Comment:
not directly an issue, but love how you mentioned it XD
I found the same thing a few months ago, it is quite hilarious... wonder if there is a good reason for it